src/core/connection.h

```diff
@@ -581,6 +588,12 @@ typedef struct QUIC_CONNECTION {
     //
     uint32_t PreviousQuicVersion;
 
+    //
+    // Initially-attempted QUIC version.
+    // Only populated during compatible version negotiation.
+    //
+    uint32_t InitialQuicVersion;
```

Comment:
We have a lot of state that increases the size of the connection object that's only used during the handshake. In future, it might be worth having a separate (lookaside list allocated) struct just for handshake state that we can free after the handshake completes. Something for the future. Not this PR.

Reply:
Fair point, the tricky part is the debugging information that could be lost by doing that. We should discuss this further.

Reply:
I created this issue https://github.com/microsoft/msquic/issues/1829 to track this further.